File: Basic/encryption-and-decryption/decode_the_message/Python3/0001_solution.py
def decrypt(color_codes):
    """Returns the decrypted message.

    Parameters
    ----------
    color_codes : list
        List of different color codes.

    Returns
    ------
    str
        The decrypted message.
    """
    result = ''
    for color in color_codes:
        c1, c2, c3 = color[:2], color[2:4], color[4:]
        result += chr(int(c1, 16)) + chr(int(c2, 16)) + chr(int(c3, 16))
    return result


def main():
    for _ in range(int(input())):
        input()
        color_codes = input().split()
        print(decrypt(color_codes))


if __name__ == "__main__":
    main()

File: Basic/encryption-and-decryption/decode_the_message/Java/0001_solution.java
import java.io.*;

class Solution {
    /**
    * This function decrypts the color codes.
    * @param color_codes This is the array of different color codes.
    * @return String This returns decrypted message.
    */
    private static String decrypt(String[] color_codes){
        StringBuilder builder = new StringBuilder();
        for (String temp : color_codes) {
            for (int i = 0; i < temp.length(); i += 2) {
                builder.append(
                    (char) Integer.parseInt(
                        temp.substring(i, i + 2), 16
                    )
                );
            }
        }
        return builder.toString();
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        while (t-- != 0) {
            Integer.parseInt(br.readLine());
            String[] str = br.readLine().split(" ");
            System.out.println(decrypt(str));
        }
    }
}